### PR TITLE
Add ImplicitExpect cop

### LINKED ISSUE #194: Enforce either `it { should ... }` OR `it { is_expected.to }`

### PR DESCRIPTION
**Pull request by backus**
Closes #194

### CODE REVIEW

**Comment by backus** [comment]
@andyw8 and @bquorning could you review this cop?

**Comment by bquorning** [comment]
As I understand it (from reading http://rspec.info/blog/2014/05/notable-changes-in-rspec-3/#new-api-for-one-liners-isexpected), setting `config.syntax = :expect` has no influence on the support for `it { should ... }` or `it { is_expected.to ... }` – since this particular `should` is not a monkey patch on `Object`. Is that correct?

File: config/default.yml [hunk 1]
```diff
@@ -113,6 +113,14 @@ RSpec/ExampleLength:
   Enabled: true
   Max: 5
 
+RSpec/ImplicitExpect:
+  Description: Check that a consist implict expectation style is used.
```
Comment (andyw8) [inline config/default.yml]: "consistent implicit"

File: lib/rubocop/cop/rspec/implicit_expect.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check that a consist implict expectation style is used.
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/implicit_expect.rb]: "consistent implicit"

Comment (backus) [inline lib/rubocop/cop/rspec/implicit_expect.rb]: Good eye!

File: config/default.yml (hunk 1 above)
Comment (backus) [inline config/default.yml]: Thanks! For future reference you only need to flag something in the description on the class. The build enforces that this description is the same

**Comment by backus** [comment]
@bquorning correct!

File: spec/rubocop/cop/rspec/implicit_expect_spec.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    end
+
+    it 'flags it { is_expected.to }' do
+      expect_violation(<<-RUBY)
+        it { is_expected.to be_truthy }
+             ^^^^^^^^^^^^^^ Prefer `should` over `is_expected.to`.
+      RUBY
+    end
+
+    it 'flags it { is_expected.to_not }' do
```
Comment (bquorning) [inline spec/rubocop/cop/rspec/implicit_expect_spec.rb]: Perhaps also add an example for `it { is_expected.not_to }`?

Comment (backus) [inline spec/rubocop/cop/rspec/implicit_expect_spec.rb]: Sure

**Comment by bquorning** [comment]
:+1:

**Pull request merged**